dotnet/src/IntegrationTests/Connectors/Weaviate/WeaviateMemoryStoreTests.cs

```diff
@@ -0,0 +1,282 @@
+﻿using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.Connectors.Memory.Weaviate;
+using Microsoft.SemanticKernel.Connectors.Memory.Weaviate.Diagnostics;
+using Microsoft.SemanticKernel.Memory;
+using Xunit;
+
+namespace SemanticKernel.IntegrationTests.Connectors.Weaviate;
+
+/// <summary>
+///     Tests for <see cref="s_weaviateMemoryStore" /> collection and upsert operations.
+/// </summary>
+[Collection("Sequential")]
+public class WeaviateMemoryStoreTests
+{
+    private static readonly HttpClient s_httpClient = new()
+    {
+        DefaultRequestHeaders =
+        {
+            { "authorization", "my-secret-key" }
+        }
+    };
+    
+    private static readonly WeaviateMemoryStore s_weaviateMemoryStore = new("http", "localhost", 8080, httpClient: s_httpClient);
+
+    [Fact(Skip = "Do not run on CI")]
```

Comment:
As devis noted, would be nice to have some breadcrumb for folks to set things up to run these tests locally.

Reply:
I have added the `docker-compose.yml` alongside these integration tests and updated the comment to instruct the user.